크레인-인형뽑기-게임.js

```diff
@@ -0,0 +1,31 @@
+function solution(board, moves) {
+    const LENBOARD = board.length;
+    let newBoard = board.map((item) => []);
+    let pushedIcon = [];
+    let score = 0;
+    for (let primary = 0; primary < LENBOARD; primary++) {
+        for (let secondary = 0; secondary < LENBOARD; secondary++) {
+            newBoard[secondary].push(board[primary][secondary]);
+        }
+    }
+    for (let selectLine in moves) {
+        for (let selectedLine = 0; selectedLine < LENBOARD; selectedLine++) {
+            if (newBoard[moves[selectLine] - 1][selectedLine]) {
+                if (
+                    pushedIcon[0] !==
+                    newBoard[moves[selectLine] - 1][selectedLine]
+                ) {
+                    pushedIcon.unshift(
+                        newBoard[moves[selectLine] - 1][selectedLine]
+                    );
```

Comment:
```suggestion
            if (newBoard[moves[selectLine] - 1][selectedLine]) {
                if (
                    pushedIcon[0] !==
                    newBoard[moves[selectLine] - 1][selectedLine]
                ) {
                    pushedIcon.unshift(
                        newBoard[moves[selectLine] - 1][selectedLine]
                    );
```
이 로직 안에서만 newBoard[moves[selectLine] - 1][selectedLine] 이 값을 3번 쓰고있기 때문에 이런 경우에는 하나의 변수로 빼서 사용하는 것이 가독성이 더 좋을거같아요